recipe/meta.yaml

```diff
@@ -24,19 +24,20 @@ source:
       - patches/0001-don-t-bake-non-relocatable-CMAKE_INSTALL_FULL_LIBDIR.patch
       # backport apache/arrow#34019 to disable useless pkgconfig search that takes ~15min
       - patches/0002-GH-33882-C-Don-t-find-.pc-files-with-ARROW_BUILD_STA.patch
+      # backport apache/arrow#34498, 34878 & #34881 for pandas 2.0 compatibility
+      - patches/0003-GH-34404-Python-Failing-tests-because-pandas.Index-c.patch
+      - patches/0004-GH-15070-Python-CI-Compatibility-with-pandas-2.0-348.patch
+      - patches/0005-GH-34880-Python-CI-Fix-Windows-tests-failing-with-la.patch
```

Comment:
Is there a reason we need to pull these into this PR? Could this be done in a separate PR?

Reply:
Yes, CI is failing without it. For the maintenance branches I just added `pandas <2` as a test constraint, but I wanted to have at least one live version that's fully compatible with pandas 2.0.

Reply:
I also don't see the point of separating things into different PRs - we already have a huge amount of (re)builds on this feedstock, so I try to condense them where reasonable. However, the commits are reasonably-separated and if we want we can easily back out stuff later.

(Both these things are in stark contrast to how upstream development works, but the same development model doesn't apply well to the feedstock IMO).

Reply:
Apologies, I didn't read the recipe close enough where I assumed there was a `pandas<2` constraint that was being removed alongside the patches. Given there isn't one this seems much more fine to me.

Reply:
No worries, thanks for the review! :)